File: src/main/java/com/extendedae_plus/network/base/BPacketGeneric.java
package com.extendedae_plus.network.base;

import net.neoforged.neoforge.network.handling.IPayloadContext;

public interface BPacketGeneric extends CPacketGeneric, SPacketGeneric {
    @Override
    default void handle(final IPayloadContext context) {
        switch (context.flow()) {
            case CLIENTBOUND -> CPacketGeneric.super.handle(context);
            case SERVERBOUND -> SPacketGeneric.super.handle(context);
        }
    }
}

File: src/main/kotlin/com/extendedae_plus/network/base/BPacketGeneric.kt
package com.extendedae_plus.network.base

import net.minecraft.network.protocol.PacketFlow
import net.neoforged.neoforge.network.handling.IPayloadContext

interface BPacketGeneric : CPacketGeneric, SPacketGeneric {
    override fun handle(context: IPayloadContext) = when (context.flow()) {
        PacketFlow.CLIENTBOUND -> super<CPacketGeneric>.handle(context)
        PacketFlow.SERVERBOUND -> super<SPacketGeneric>.handle(context)
    }
}
